From 648be44feeca41adeb9b16ca2d78752e91078929 Mon Sep 17 00:00:00 2001
From: ltian <le.tian@windriver.com>
Date: Thu, 11 Nov 2010 15:56:59 +0800
Subject: [PATCH 018/132] Cavium: Update cavium-octeon Kconfig options

Source: SDK 2.0.0-366

SDK 2.0 adds support for OCTEON2 and some additional CVMX parameter
checking options. Update the Kconfig file to make provisions to enable
these options.

Signed-off-by: ltian <le.tian@windriver.com>
---
 arch/mips/Kconfig               |    1 +
 arch/mips/cavium-octeon/Kconfig |   38 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 39 insertions(+), 0 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 9941ca8..e171a7a 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -649,6 +649,7 @@ config CAVIUM_OCTEON_SIMULATOR
 	select SYS_SUPPORTS_BIG_ENDIAN
 	select SYS_SUPPORTS_HIGHMEM
 	select SYS_SUPPORTS_HOTPLUG_CPU
+	select SYS_HAS_EARLY_PRINTK
 	select SYS_HAS_CPU_CAVIUM_OCTEON
 	help
 	  The Octeon simulator is software performance model of the Cavium
diff --git a/arch/mips/cavium-octeon/Kconfig b/arch/mips/cavium-octeon/Kconfig
index 8c188a2..13792c0 100644
--- a/arch/mips/cavium-octeon/Kconfig
+++ b/arch/mips/cavium-octeon/Kconfig
@@ -3,6 +3,44 @@ config CAVIUM_OCTEON_SPECIFIC_OPTIONS
 	depends on CPU_CAVIUM_OCTEON
 	default "y"
 
+config CAVIUM_OCTEON2
+       bool "Build the kernel to be used only on OCTEON II processor cores"
+       depends on CPU_CAVIUM_OCTEON
+       default "n"
+       help
+         This option enables the generation of Octeon2 specific
+         instructions by the compiler, resulting in a kernel that is
+         more efficient, but that will not run on Octeon and
+         OcteonPlus processor cores.
+
+
+config CAVIUM_OCTEON_CHK_CVMX_PARAMETER
+       bool "Build the kernel using -DCVMX_ENABLE_PARAMETER_CHECKING=1"
+       depends on CPU_CAVIUM_OCTEON
+       default "n"
+       help
+         Compile the kernel with CVMX parameter checking enabled.
+         This might catch some programming errors, but will result in
+         a slower kernel.
+
+config CAVIUM_OCTEON_CHK_CVMX_ADDRESS
+       bool "Build the kernel using -DCVMX_ENABLE_CSR_ADDRESS_CHECKING=1"
+       depends on CPU_CAVIUM_OCTEON
+       default "n"
+       help
+         Compile the kernel with CVMX CSR address checking enabled.
+         This might catch some programming errors, but will result in
+         a slower kernel.
+
+config CAVIUM_OCTEON_CHK_CVMX_POW
+       bool "Build the kernel using -DCVMX_ENABLE_POW_CHECKS=1"
+       depends on CPU_CAVIUM_OCTEON
+       default "n"
+       help
+         Compile the kernel with CVMX CSR POW checking enabled.
+         This might catch some programming errors, but will result in
+         a slower kernel.
+
 config CAVIUM_OCTEON_2ND_KERNEL
 	bool "Build the kernel to be used as a 2nd kernel on the same chip"
 	depends on CAVIUM_OCTEON_SPECIFIC_OPTIONS
-- 
1.6.5.2

